### ISSUE TITLE
`doc_markdown` false negative for inlined code blocks

### ISSUE THREAD

**Issue by ebobrow** [rust-lang/rust-clippy#7427]
Lint name: doc_markdown

I tried this code:

```rust
#![warn(clippy::doc_markdown)]
/// - ``` fn main() { todo!() } ```
fn main() {}
```

I expected to see this happen:
This should be linted for needless main in a doctest.

Instead, this happened:
Clippy didn't lint. It seems that inlined code blocks are parsed as `Code` instead of `CodeBlock`s by `pulldown_cmark`, so clippy doesn't lint them as it should.

### Meta

- `cargo clippy -V`: e.g. clippy 0.0.212 (f455e46 2020-06-20)
- `rustc -Vv`:
  ```
  rustc 1.46.0-nightly (f455e46ea 2020-06-20)
  binary: rustc
  commit-hash: f455e46eae1a227d735091091144601b467e1565
  commit-date: 2020-06-20
  host: x86_64-unknown-linux-gnu
  release: 1.46.0-nightly
  LLVM version: 10.0
  ```

**Comment by Alexendoo** [comment]
This is how rustdoc sees it too, it renders it as an inline code block and it doesn't get doctested
